core/lib/dal/src/blocks_dal.rs

```diff
@@ -2273,6 +2295,48 @@ impl BlocksDal<'_, '_> {
         .map(|row| row.tree_writes_are_present)
         .unwrap_or(false))
     }
+
+    pub(crate) async fn get_l2_to_l1_logs_for_batch<L>(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> DalResult<Vec<L>>
+    where
+        L: From<StorageL2ToL1Log>,
+    {
+        let results = sqlx::query_as!(
+            StorageL2ToL1Log,
+            r#"
+            SELECT
+                miniblock_number,
+                log_index_in_miniblock,
+                log_index_in_tx,
+                tx_hash,
+                l1_batch_number,
+                shard_id,
+                is_service,
+                tx_index_in_miniblock,
+                tx_index_in_l1_batch,
+                sender,
+                key,
+                value
+            FROM
+                l2_to_l1_logs
+                JOIN miniblocks ON l2_to_l1_logs.miniblock_number = miniblocks.number
+            WHERE
+                l1_batch_number = $1
+            ORDER BY
+                tx_index_in_l1_batch,
+                log_index_in_tx
```

Comment:
Dumb question: Is ordering by `miniblock_number, log_index_in_miniblock` (i.e., the primary key of `l2_to_l1_logs`) the same or different? Intuitively, if they are the same, ordering by the primary key could be more efficient.

Reply:
I have tested locally. Ordering by `key` does not work.

Reply:
ah nice!
changed